### PR TITLE
DSDEGP-1910: Sort files in Status Page according to wireframes

### PR DESCRIPTION
**Pull request by vvicario**
### Description

- Sort files in Status Page according to wireframes
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any JIRA issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._

----

### Checklist

- [X] **Submitter**: Include the JIRA issue number in the PR description
- [X] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=189431877).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: src/app/file-manager/reducers/transferables.reducer.ts [hunk 1]
```diff
@@ -38,6 +38,58 @@ export class TransferableState {
         return this.items.filter(x => x.type === Type.UPLOAD).length;
     }
 }
+function sortList(items) {
```
Comment (pshapiro4broad) [inline src/app/file-manager/reducers/transferables.reducer.ts]: should declare `items` as type `Item[]`

File: src/app/file-manager/reducers/transferables.reducer.ts [hunk 2]
```diff
@@ -38,6 +38,58 @@ export class TransferableState {
...
+    items.sort((a, b) => {
+        switch (a.status) {
+          case ItemStatus.DOWNLOADING:
+          {
```
Comment (pshapiro4broad) [inline src/app/file-manager/reducers/transferables.reducer.ts]: pretty sure typescript doesn't require a `{` `}` around the case block here, e.g.
```
case ItemStatus.DOWNLOADING:
    if (b.status === a.status) {
        ...
```

File: src/app/file-manager/reducers/transferables.reducer.ts [hunk 3]
```diff
@@ -99,11 +151,13 @@ export function TransferablesReducer(state = initialState, action: Action): Tran
 
         case TransferablesActions.ADD_ITEM:
             const add_item = [...state.items, action.payload];
+            sortList(add_item);
```
Comment (pshapiro4broad) [inline src/app/file-manager/reducers/transferables.reducer.ts]: I know it will result in sorting the list in a few cases where sorting isn't necessary, but I think it would be better to sort the list in the constructor for `TransferableState`. That avoids the duplicate code here and better documents that the list is always sorted.

File: src/app/file-manager/warning-modal/warning-modal.component.html [hunk 4]
```diff
@@ -9,7 +9,7 @@ <h2 mat-dialog-title>
 <mat-dialog-content>
   <div class="flex-container">
     <p>All current
-      <span *ngIf="data == 'exit' || data == 'logout'">transfers</span>
+      <span *ngIf="data == 'quit' || data == 'logout'">transfers</span>
```
Comment (pshapiro4broad) [inline src/app/file-manager/warning-modal/warning-modal.component.html]: Is this an unrelated change?

**Comment by dags** [comment]
This was done this way to keep all State changes confined to the reducer code. To let other sort options we could just send sort configuration info to reducer as action. But we can review this approach after alpha release of course.Regards.Diego.El 3 ene. 2018 18:40, vvicario <notifications@github.com> escribió:Description
Sort files in Status Page according to wireframes
Explain the motivation for making this change. What existing problem does the pull request solve?
Mention any JIRA issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos.

Checklist
 Submitter: Include the JIRA issue number in the PR description Submitter: Check documentation and code comments. Add explanatory PR comments if helpful. Submitter: Ensure that you have added or updated tests Submitter: If you're adding/switching libraries or otherwise changing the design, update the design documentation. Submitter: JIRA ticket checks:
Acceptance criteria exists and is metNote any changes to implementation from the descriptionAdd notes on what you've tested

Review cycle:
Team may comment on PR at willReviewer assigns to submitter for feedback fixesSubmitter rebases to develop again if necessarySubmitter makes further commits. DO NOT SQUASHSubmitter updates documentation as neededSubmitter reassigns to reviewer for further feedback

 Submitter: Verify all tests go green, including CI tests Submitter: Squash commits and merge to develop Submitter: Delete branch after merge Submitter: Test this change works on dev environment after deployment. YOU own getting it fixed if dev isn't working for ANY reason! Submitter: Move the JIRA issue to QA once this checklist is completed



You can view, comment on, or merge this pull request online at:
  https://github.com/broadinstitute/firecloud-explorer/pull/39

Commit Summary
sortingMerge branch 'develop' of https://github.com/broadinstitute/firecloud-explorer into trk-DSDEGP-1910-sortItemsfixminorssort by casesort listMerge branch 'develop' of https://github.com/broadinstitute/firecloud-explorer into trk-DSDEGP-1910-sortItemsorder by cancel

File Changes

    M
    src/app/file-manager/reducers/transferables.reducer.ts
    (91)
  
    M
    src/app/file-manager/warning-modal/warning-modal.component.html
    (6)
  

Patch Links:
https://github.com/broadinstitute/firecloud-explorer/pull/39.patchhttps://github.com/broadinstitute/firecloud-explorer/pull/39.diff

File: src/app/file-manager/warning-modal/warning-modal.component.html (hunk 4 above)
Comment (LeoForconesi) [reply src/app/file-manager/warning-modal/warning-modal.component.html]: Phil, this change is a quick fix for develop, we can move this fix to a new branch if you think is better.

**Comment by vvicario** [comment]
@pshapiro4broad It's ready to review.

File: src/app/file-manager/warning-modal/warning-modal.component.html (hunk 4 above)
Comment (pshapiro4broad) [reply src/app/file-manager/warning-modal/warning-modal.component.html]: It's ok just wanted to make sure it wasn't an accidental inclusion
